src/Core/Domain/Carrier/ValueObject/ShippingMethod.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Carrier\ValueObject;
+
+use PrestaShop\PrestaShop\Core\Domain\Carrier\Exception\CarrierConstraintException;
+
+/**
+ * Shipping method value for Carriers
+ */
+class ShippingMethod
+{
+    /**
+     * Use weight to calculate shipping cost
+     */
+    public const BY_WEIGHT = 1;
+
+    /**
+     * Use price to calculate shipping cost
+     */
+    public const BY_PRICE = 2;
+
+    /**
+     * A list of available values
+     */
+    public const AVAILABLE_VALUES = [
+        self::BY_WEIGHT,
```

Comment:
From the Carrier ObjectModek it seems there are two other possible values for shipping method:
- `SHIPPING_METHOD_FREE` so far I seems this one is used dyanmically when `is_free` is true, but some legacy code still depend on it, maybe we could maintain `shipping_method` automatically updated to this value when `is_free` is turned on?
- `SHIPPING_METHOD_DEFAULT` this seems to avoid defining the shipping method and instead uses the default used one based on the `PS_SHIPPING_METHOD` global configuration, I don't know if it's relevant because it seems to be used for retro compatibility, when we edit a carrier we shouldn't have this option available because you can't define ranges based on a configuration you don't know and that can change

TLDR; maybe these to extra options are not adapted any more and should be ignored in the CQRS layer?

Reply:
- `SHIPPING_METHOD_FREE` is only when `is_free = true` and `shipping_method` value is set to `SHIPPING_METHOD_FREE` automatically when we make a `getShippingMethod` in Carrier object model. We must always DON'T save this value in database. (think about edit a Carrier with `is_free` off then...)
- `SHIPPING_METHOD_DEFAULT`: As you wish, but it looks like for retro compatibility so....

Reply:
`PS_SHIPPING_METHOD` configuration key is some really old thing that was not even used in 1.6, so it can be safely removed.

I doubt that there is some PS installation where there is a carrier with SHIPPING_METHOD_DEFAULT anymore...

Ping @kpodemski

Reply:
Digged deeper, the config is unused since 1.5.5.0, when the carrier wizard was introduced. 👍

https://github.com/PrestaShop/PrestaShop/blob/1.5.5.0/controllers/admin/AdminCarrierWizardController.php

Reply:
@boherm then maybe we should clean the `SHIPPING_METHOD_DEFAULT` completely
About the free shipping method, should we handle its value dynamically in the QueryResult as well then?

Reply:
@jolelievre, we must avoid to do that in the EditableCarrier class, we need here the real value in database.